internal/query/query.go

```diff
@@ -108,6 +109,10 @@ func (q *Queries) Health(ctx context.Context) error {
 	return q.client.Ping()
 }
 
+func (q *Queries) SetPermissionCheck(permissionCheck func(ctx context.Context, permission string, orgID string, resourceID string) (err error)) {
+	q.checkPermission = permissionCheck
+}
```

Comment:
Isn't it better if the constructor takes the `permissionCheck` function instead? Otherwise, we might end up with `nil` function bugs if we `StartQueries` somewhere and forget to call `SetPermissionCheck`.

It is also acceptable if we set a no-op function as default in `StartQueries` to at least prevent the function being `nil`.

Reply:
found a solution to have in the constructor